admin/used_flusher.go

```diff
@@ -0,0 +1,84 @@
+package admin
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rilldata/rill/admin/database"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"go.uber.org/zap"
+)
+
+type usedFlusher struct {
+	deployments map[string]bool
+	lock        sync.Mutex
+	db          database.DB
+	logger      *zap.Logger
+	ctx         context.Context
+	cancel      context.CancelFunc
+}
+
+func newUsedFlusher(logger *zap.Logger, db database.DB) *usedFlusher {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	used := &usedFlusher{
+		deployments: make(map[string]bool),
+		db:          db,
+		logger:      logger,
+		ctx:         ctx,
+		cancel:      cancel,
+	}
+	go used.runBackground(ctx)
+
+	return used
+}
+
+func (u *usedFlusher) Deployment(id string) {
+	u.lock.Lock()
+	defer u.lock.Unlock()
+
+	u.deployments[id] = true
+}
+
+func (u *usedFlusher) runBackground(ctx context.Context) {
+	ticker := time.NewTicker(20 * time.Second)
+	go func() {
```

Comment:
Can remove the `go func()` scope here, since you're already calling `go used.runBackground`, so it will already run in the background